pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -125,15 +114,22 @@ var _ = Describe("podNIC", func() {
 		})
 		Context("and starting the DHCP server fails", func() {
 			BeforeEach(func() {
-				mockNetwork.EXPECT().StartDHCP(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("podnic_test: forcing failure at DHCP start"))
+				podnic.dhcpConfigurator = &fakeDHCPConfigurator{
+					shouldEnsureDHCPFail: true,
+				}
 			})
 			It("phase2 should panic", func() {
 				Expect(func() { podnic.PlugPhase2(domain) }).To(Panic())
 			})
 		})
 		Context("and starting the DHCP server succeed", func() {
 			BeforeEach(func() {
-				mockNetwork.EXPECT().StartDHCP(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1)
+				podnic.dhcpConfigurator = &fakeDHCPConfigurator{
+					shouldGenerateFail: false,
+					dhcpConfig:         &cache.DHCPConfig{},
+					podInterfaceName:   primaryPodInterfaceName,
+					dhcpOptions:        vmi.Spec.Domain.Devices.Interfaces[0].DHCPOptions,
```

Comment:
would it make sense to assert that `vmi.Spec.Domain.Devices.Interfaces` is not empty ?